docs/src/pages/system/basics/Why.js

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import TrendingUpIcon from '@material-ui/icons/TrendingUp';
+import Box from '@material-ui/core/Box';
+
+export default function Why() {
+  return (
+    /* prettier-ignore */
```

Comment:
For demos we should rely on prettier. Any existing prettier-ignore in the codebase is no reason to use it more.

Reply:
@oliviertassinari was the intention here to shorten the Demo to showcase the benefit of `sx` vs `styled()`?

Reply:
The objective was to shorten the demo, to act on this aspect: https://twitter.com/timolins/status/1295736207427731458. 

It's the same incentive for exploring this direction, at its core, it's about how prettier auto-format the content:

<img width="599" alt="Capture d’écran 2020-11-08 à 21 05 23 (2)" src="https://user-images.githubusercontent.com/3165635/98826538-1ee39f80-2436-11eb-8c27-589d1cb18c4b.png">

```jsx
'display:flex flex-direction:column py:4 py:sm:5'.split(' ').reduce((acc, item) => {
  const split = item.split(':');
  if (split.length === 3) {
    const [property, breakpoint, value] = split;
    if (typeof acc[property] === 'string') {
      acc[property] = {
        xs: acc[property],
        [breakpoint]: value,
      };
    } else if (typeof acc[property] === 'object') {
      acc[property] = {
        ...acc[property],
        [breakpoint]: value,
      };
    } else {
      acc[property] = {
        [breakpoint]: value,
      };
    }
  } else {
    const [property, value] = split;
    acc[property] = value;
  }
  return acc;
}, {});
```

Reply:
The demo is just not shorter. It's just your opinionated formatting and I don't think we're in the business of giving formatting opinions.

Reply:
@eps1lon By "demo", do you mean the demo in the documentation or the string syntax in the screenshot?
Right, we can be more precise. It's intrinsically the same length so it's not shorter per say. However, it's vertically shorter, which matters.

Reply:
The line I'm adressing. We use prettier for formatting. There's a reason the whole ecosystem has switched!

Reply:
I believe the switch is about two main reasons: 70%: 1. we don't have time to format by hand, 30% 2. it's simpler to have everything formatted the same way.

Reply:
> ```
> 'display:flex flex-direction:column py:4 py:sm:5'.split(' ').reduce((acc, item) => {
>   const split = item.split(':');
>   if (split.length === 3) {
>     const [property, breakpoint, value] = split;
>     if (typeof acc[property] === 'string') {
>       acc[property] = {
>         xs: acc[property],
>         [breakpoint]: value,
>       };
>     } else if (typeof acc[property] === 'object') {
>       acc[property] = {
>         ...acc[property],
>         [breakpoint]: value,
>       };
>     } else {
>       acc[property] = {
>         [breakpoint]: value,
>       };
>     }
>   } else {
>     const [property, value] = split;
>     acc[property] = value;
>   }
>   return acc;
> }, {});
> ```

Will create draft PR for this syntax tomorrow, haven't had a chance to do it till now. 

Regarding prettier, should we maybe just try to increase the line length we support for the demos? For just two, or three props in the sx I believe they should be kept on the same line 🤷

Reply:
Increasing the line length leads to scroll-overflow and less readability.